snuba/utils/codecs.py

```diff
@@ -0,0 +1,28 @@
+from abc import ABC, abstractmethod
+from typing import Generic, TypeVar
+
+
+TEncoded = TypeVar("TEncoded")
+
+TDecoded = TypeVar("TDecoded")
+
+
+class Codec(Generic[TEncoded, TDecoded], ABC):
```

Comment:
Should this become a more abstract entity than a codec for Kafka messages ? If not then it should probably be somewhere in the consumer package.

Reply:
It's not inherently limited to the consumer implementation, but I also don't have a concrete use case for it elsewhere. I'm fine with moving it.

Reply:
Moved this into `snuba.utils.streams.codecs` here: 194585b7a2b1b2ca18819e2fc2328d049c72b5cc